### PR TITLE
Display private images in WYSIWYG editor

### LINKED ISSUE #5920: Access token not being appended to images uploaded to WYSIWYG field
# What's going wrong:

Uploading an image in the WYSIWYG field results in a broken link.

When opening the broken image in a new tab, I get this error:

```
{"errors":[{"message":"You don't have permission to access this.","extensions":{"code":"FORBIDDEN"}}]}
```

I expected the image to appear in the WYSIWYG field instead of a broken image link symbol.

The image is uploading properly, it's just the link that is broken. I know this because I can take the access token from the URL of another image and paste it at the end of the WYSIWYG image's URL.

**It appears that the access token is not being properly appended to images uploaded through the WYSIWYG editor.**

Image uploads work everywhere else in Directus, including updating the project logo, in dedicated Image fields, etc. I tested `local` and `s3` storage options and the issue occurs with both.

## Reproducing the issue:

Attempt to upload an image in a WYSIWYG text field.

To at least access the image, right click a working image in Directus and `open image in new tab`. Find the query string in the URL that begins with `?access_token=`. Copy that entire string. Go back to the broken image in the WYSIWYG editor and `open image in new tab`. Paste the copied query string at the end of the URL and press enter. The image will load.

## Environment:

- Directus v9.0.0-rc.69
- Postgres 12
- Docker
 - Chrome 90.0.4430.212

## Other variables:

- I'm using S3 buckets through both Wasabi and DigitalOcean.

### PR DESCRIPTION
**Pull request by jamescammarano**
Fixes #5920 
Appends an access token to the image URL when displaying it in the WYSIWYG editor as well as in the image upload drawer. 
Removes access token before adding to the database.

### CODE REVIEW

File: app/src/interfaces/input-rich-text-html/input-rich-text-html.vue [hunk 1]
```diff
@@ -262,11 +264,45 @@ export default defineComponent({
...
+					);
+				}
 				return props.value;
 			},
+
 			set(newValue: string) {
 				if (newValue !== props.value && (props.value === null && newValue === '') === false) {
-					emit('input', newValue);
+					const url = getPublicURL();
+					const remove_token = newValue.replace(
```
Comment (rijkvanzanten) [inline app/src/interfaces/input-rich-text-html/input-rich-text-html.vue:290]: Use camelCase

File: app/src/interfaces/input-rich-text-html/input-rich-text-html.vue [hunk 2]
```diff
@@ -262,11 +264,45 @@ export default defineComponent({
...
+								url
+							)}assets/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}(?:\\?[^#"]+)?(?:#[^"]*)?)("[^>]*>)`,
+							'gi'
+						),
+						(_, pre, matchedUrl, post) => {
+							const matched = new URL(matchedUrl);
+							const params = new URLSearchParams(matched.search);
+							params.delete('amp;access_token');
+							params.delete('access_token');
+							return `${pre}${matched.origin}${matched.pathname}${params}${post}`;
```
Comment (rijkvanzanten) [inline app/src/interfaces/input-rich-text-html/input-rich-text-html.vue:302]: We still have to include the configured token when it exists (from `props.imageToken`)

File: app/src/interfaces/input-rich-text-html/input-rich-text-html.vue [hunk 3]
```diff
@@ -409,3 +445,5 @@ export default defineComponent({
 	font-size: 24px;
 }
 </style>
+
+function getPublicPath() { throw new Error('Function not implemented.'); }
```
Comment (rijkvanzanten) [inline app/src/interfaces/input-rich-text-html/input-rich-text-html.vue:449]: This doesn't seem right here

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

File: app/src/interfaces/input-rich-text-html/input-rich-text-html.vue [hunk 4]
```diff
@@ -299,10 +299,12 @@ export default defineComponent({
 							const params = new URLSearchParams(matched.search);
 							params.delete('amp;access_token');
 							params.delete('access_token');
-							return `${pre}${matched.origin}${matched.pathname}${params}${post}`;
+							return `${pre}${matched.origin}${matched.pathname}${
+								props.imageToken ? props.imageToken : ''
+							}${params}${post}`;
```
Comment (rijkvanzanten) [inline app/src/interfaces/input-rich-text-html/input-rich-text-html.vue:304]: Wouldn't this resolve into `https://localhost:8055/assets/abc-def-ghi/my-custom-image-token?query=ab` ? The imageToken is part of params, so I think it has to be

```
params.set('access_token', props.imageToken);

return `${pre}${matched.origin}${matched.pathname}${params}${post}`;

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

**Review by rijkvanzanten — APPROVED** [review]

**Review by rijkvanzanten — APPROVED** [review]

**Comment by domdus** [comment]
Hey guys,

I'm on rc.95 and not sure if this is fully working. I can embed see an image in WYSIWYG editor at first.
I also see the token being added when right-click the image -> image (modal view): 
image

If you **instead** select the image and click "image" from the toolbar on top (sidebar view), the token is added **twice** and images goes HTTP 500:

**Request-URL:**
`https://<DOMAIN>/assets/159ce235-9b5f-4dee-9d48-ff8de56a7509?access_token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImM0NjZkMDQ3LWQwZGItNDliZC1iN2EyLWQ3NmQzMjMzZDdjMyIsImlhdCI6MTYzMzY5MjE4MSwiZXhwIjoxNjMzNjkzMDgxLCJpc3MiOiJkaXJlY3R1cyJ9.KIpMiYa1WiblxDLyEXqLn1cKSJh0YlM-03mIqBMfzMk&access_token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImM0NjZkMDQ3LWQwZGItNDliZC1iN2EyLWQ3NmQzMjMzZDdjMyIsImlhdCI6MTYzMzY5Mjc5NSwiZXhwIjoxNjMzNjkzNjk1LCJpc3MiOiJkaXJlY3R1cyJ9.hIZoQhT4RfJ2FJxZ_5IpT-4z7R6IkgWL4B8iSE4Mt5c`

image

**Comment by domdus** [comment]
#8653

**Pull request merged**